integration-cli/docker_cli_build_test.go

```diff
@@ -243,6 +243,7 @@ func TestBuildEnvironmentReplacementEnv(t *testing.T) {
   FROM scratch
   ENV foo foo
   ENV bar ${foo}
+  ENV env1=$foo env2=${env1} env3="$env2" env4="${env3}"
```

Comment:
@duglin Can we add a test case for `'$env'` and make sure it does not get substituted?